op-chain-ops/cmd/withdrawal/prove.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/wait"
+	"github.com/ethereum-optimism/optimism/op-node/bindings"
+	bindingspreview "github.com/ethereum-optimism/optimism/op-node/bindings/preview"
+	"github.com/ethereum-optimism/optimism/op-node/withdrawals"
+	op_service "github.com/ethereum-optimism/optimism/op-service"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/ethclient/gethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/lmittmann/w3"
+	"github.com/urfave/cli/v2"
+)
+
+var (
+	L1Flag = &cli.StringFlag{
+		Name:    "l1",
+		Usage:   "HTTP provider URL for L1.",
+		EnvVars: op_service.PrefixEnvVar(EnvVarPrefix, "L1"),
+	}
+	TxFlag = &cli.StringFlag{
+		Name:    "tx",
+		Usage:   "Transaction hash of withdrawal on L2",
+		EnvVars: op_service.PrefixEnvVar(EnvVarPrefix, "TX"),
+	}
+	PortalAddressFlag = &cli.StringFlag{
+		Name:    "portal-address",
+		Usage:   "Address of the optimism portal contract.",
+		EnvVars: op_service.PrefixEnvVar(EnvVarPrefix, "PORTAL_ADDRESS"),
+	}
+)
+
+func ProveWithdrawal(ctx *cli.Context) error {
+	logger, err := setupLogging(ctx)
+	if err != nil {
+		return err
+	}
+
+	txMgr, err := createTxMgr(ctx, logger, L1Flag.Name)
+	if err != nil {
+		return err
+	}
+
+	txHash := common.HexToHash(ctx.String(TxFlag.Name))
+	if txHash == (common.Hash{}) {
+		return errors.New("must specify tx hash")
+	}
+
+	rpcClient, err := rpc.DialContext(ctx.Context, ctx.String(L2Flag.Name))
+	if err != nil {
+		return fmt.Errorf("failed to connect to L2: %w", err)
+	}
+	proofClient := gethclient.New(rpcClient)
+	l2Client := ethclient.NewClient(rpcClient)
+
+	l1Client, err := ethclient.DialContext(ctx.Context, ctx.String(L1Flag.Name))
+	if err != nil {
+		return fmt.Errorf("failed to connect to L1: %w", err)
+	}
+
+	rcpt, err := l2Client.TransactionReceipt(ctx.Context, txHash)
+	if err != nil {
+		return fmt.Errorf("failed to get transaction receipt: %w", err)
+	}
+
+	portalAddr := common.HexToAddress(ctx.String(PortalAddressFlag.Name))
+	portal, err := bindingspreview.NewOptimismPortal2(portalAddr, l1Client)
+	if err != nil {
+		return fmt.Errorf("failed to bind portal: %w", err)
+	}
+	factoryAddr, err := portal.DisputeGameFactory(&bind.CallOpts{Context: ctx.Context})
+	if err != nil {
+		return fmt.Errorf("failed to fetch dispute game factory address from portal: %w", err)
+	}
+
+	factory, err := bindings.NewDisputeGameFactoryCaller(factoryAddr, l1Client)
+	if err != nil {
+		return fmt.Errorf("failed to bind dispute game factory: %w", err)
+	}
+
+	_, err = wait.ForGamePublished(ctx.Context, l1Client, portalAddr, factoryAddr, rcpt.BlockNumber)
```

Comment:
nit: This is OK for devnets. But worth noting that the game being waited on may be incorrect. `wait.ForGamePublished` doesn't ensure that the proposal root is valid.

Reply:
Yeah I figured it was better to live with that since it will nearly always work and otherwise we'd need to also supply an op-node endpoint.  If the output isn't valid generating the proof should fail which would be a bit confusing but not the end of the world.